hi-ml-azure/src/health/azure/azure_util.py

```diff
@@ -476,27 +556,148 @@ def get_latest_aml_runs_from_experiment(args: Namespace, workspace: Workspace) -
+    :param aml_workspace: Optional Azure ML Workspace object
+    :param workspace_config_path: Optional path to a Workspace config file
+    :return: The Azure ML Run object with the given run_id
+    """
+    def _get_aml_run_from_runid(run_id: str, workspace: Workspace) -> Run:  # pragma: no cover
+        """
+        Attempt to retrieve a Run from a given Workspace, specified by run_id.
+
+        :param run_id: the run_id to return
+        :param workspace: the Workspace to search in
+        :return: An Azure ML Run object
+        """
+        try:
+            run = workspace.get_run(run_id)
+        except HttpOperationError as e:
+            if 'was not found' in e.message:
+                raise ValueError(f"Run {run_id} was not found in workspace {workspace.name}. "
+                                 f"Please check the run_id and the workspace definition")
+        return run
+
+    workspace = get_workspace(aml_workspace=aml_workspace, workspace_config_path=workspace_config_path)
+    run = _get_aml_run_from_runid(run_id, workspace)
+    return run
+
+
+def get_aml_run_from_run_id_args(args: Namespace, aml_workspace: Optional[Workspace] = None,
+                                 workspace_config_path: Optional[Path] = None) -> Run:
+    """
+    Lookup the run_id arg and then retrieve the Azure ML Run object with this id.
+
+    :param args: Command line args
+    :param aml_workspace: Optional Azure ML Workspace object
+    :param workspace_config_path: Optional path to a Workspace config file
+    :return: The Azure ML Run object with the id as specified by args.run_id
+    """
+    def _get_run_id_from_args(args: Namespace) -> str:
+        """
+        Retrieve the value for run_id from the args as long as one is supplied.
+
+        :param args: The command line arguments
+        :return: A string representing the run_id as passed in to the command line
+        """
+        if not args.run_id:
+            raise ValueError("Expected to find run_id in args but did not")
+        else:
+            return args.run_id
+
+    run_id = _get_run_id_from_args(args)
+    return get_aml_run_from_run_id(run_id, aml_workspace=aml_workspace, workspace_config_path=workspace_config_path)
+
+
+def get_aml_runs_from_run_ids(args: Namespace, aml_workspace: Optional[Workspace] = None,
+                              workspace_config_path: Optional[Path] = None) -> List[Run]:
     """
     Retrieve AzureML Runs for each of the Run Ids specified in args.
 
     :param args: command line args including experiment name and number of runs to return
-    :param workspace: AML Workspace
+    :param aml_workspace: Optional Azure ML Workspace object
+    :param workspace_config_path: Optional path containing AML Workspace settings
     :return: List of AML Runs
     """
-    runs = [workspace.get_run(r_id) for r_id in args.run_ids]
+    def _get_run_ids_from_args(args: Namespace) -> List[str]:  # pragma: no cover
```

Comment:
worth having that function?

Reply:
I think so, because the default for this arg is an empty list, so this function would return an empty list without complaining